### PR TITLE
Add pid to climber winch

### PR DESCRIPTION
**Pull request by Yoel-Perman-Brilliant**

### CODE REVIEW

File: src/main/java/frc/robot/OI.java [hunk 1]
```diff
@@ -39,9 +40,18 @@ public OI() {
         xbox.getRBButton().whenPressed(new MoveGenericSubsystem(IntakePlacer.getInstance(), IntakePlacer.MAX_SPEED));
         xbox.getLTButton().whileActiveOnce(new ReleaseCargo());
 
-        xbox.getGreenButton().whenPressed(new MoveGenericSubsystem(climberWinch, ClimberWinch.DOWN_SPEED));
-        xbox.getYellowButton().whenPressed(new MoveGenericSubsystem(climberWinch, ClimberWinch.UP_SPEED));
+        //lowers the climber
+        xbox.getGreenButton().whenPressed(
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/OI.java:44]: don't do pid for moving the winch down

File: src/main/java/frc/robot/commands/ShootToHigherHub.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import com.spikes2212.dashboard.RootNamespace;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.subsystems.IntakeToTransfer;
+import frc.robot.subsystems.Transfer;
+
+import java.util.function.Supplier;
+
+public class ShootToHigherHub extends SequentialCommandGroup {
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/commands/ShootToHigherHub.java:12]: delete

File: src/main/java/frc/robot/subsystems/IntakePlacer.java [hunk 3]
```diff
@@ -8,6 +8,8 @@
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import frc.robot.RobotMap;
 
+import java.util.function.Supplier;
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/subsystems/IntakePlacer.java:11]: remove this import

File: src/main/java/frc/robot/subsystems/IntakePlacer.java [hunk 4]
```diff
@@ -17,9 +19,10 @@
 
     public static final double MAX_SPEED = 0.5;
     public static final double MIN_SPEED = -0.1;
-    public static final double SERVO_START_ANGLE = 0;
-    public static final double SERVO_TARGET_ANGLE = 20;
+    public static final double SERVO_START_ANGLE = 90;
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/subsystems/IntakePlacer.java:22]: change to 110

File: src/main/java/frc/robot/subsystems/IntakePlacer.java [hunk 5]
```diff
@@ -17,9 +19,10 @@
...
+    public static final double SERVO_TARGET_ANGLE = 0;
 
+    public final Supplier<Double> TARGET = rootNamespace.addConstantDouble("TARGET", 90);
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/subsystems/IntakePlacer.java:25]: delete this

File: src/main/java/frc/robot/subsystems/IntakePlacer.java [hunk 6]
```diff
@@ -69,6 +72,7 @@ public void configureDashboard() {
         rootNamespace.putData("move intake up", new MoveGenericSubsystem(this, MAX_SPEED));
         rootNamespace.putData("move servo", new InstantCommand(() -> setServoAngle(SERVO_TARGET_ANGLE)));
         rootNamespace.putData("reset servo", new InstantCommand(() -> setServoAngle(SERVO_START_ANGLE)));
+        rootNamespace.putData("move servo to target", new InstantCommand(() -> setServoAngle(TARGET.get())));
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/subsystems/IntakePlacer.java:75]: delete this

File: src/main/java/frc/robot/OI.java [hunk 7]
```diff
@@ -39,9 +40,18 @@ public OI() {
...
+                new MoveGenericSubsystemWithPID(climberWinch, climberWinch.ENCODER_DOWN_LIMIT,
+                        climberWinch::getPosition, climberWinch.getPIDSettings(), climberWinch.getFFSettings()));
+
+        //raises the climber
+        xbox.getYellowButton().whenPressed(
+                new MoveGenericSubsystemWithPID(climberWinch, climberWinch.ENCODER_UP_LIMIT,
+                        climberWinch::getPosition, climberWinch.getPIDSettings(), climberWinch.getFFSettings()));
         xbox.getBlueButton().whenPressed(new MoveGenericSubsystem(climberWinch, 0));
+
+        //stops the climber
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/OI.java:54]: ???

File: src/main/java/frc/robot/OI.java [hunk 8]
```diff
@@ -39,9 +40,18 @@ public OI() {
...
+        //lowers the climber
+        xbox.getGreenButton().whenPressed(
+                new MoveGenericSubsystemWithPID(climberWinch, climberWinch.ENCODER_DOWN_LIMIT,
+                        climberWinch::getPosition, climberWinch.getPIDSettings(), climberWinch.getFFSettings()));
+
+        //raises the climber
+        xbox.getYellowButton().whenPressed(
+                new MoveGenericSubsystemWithPID(climberWinch, climberWinch.ENCODER_UP_LIMIT,
+                        climberWinch::getPosition, climberWinch.getPIDSettings(), climberWinch.getFFSettings()));
         xbox.getBlueButton().whenPressed(new MoveGenericSubsystem(climberWinch, 0));
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/OI.java:52]: add documentation

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/IntakePlacer.java (hunk 3 above)
Comment (Yoel-Perman-Brilliant) [reply src/main/java/frc/robot/subsystems/IntakePlacer.java:11]: but it's used

Comment (OfriRosenbaum) [reply src/main/java/frc/robot/subsystems/IntakePlacer.java:11]: no.

File: src/main/java/frc/robot/OI.java [hunk 9]
```diff
@@ -39,9 +40,17 @@ public OI() {
...
+
+        //raises the climber
+        xbox.getYellowButton().whenPressed(
+                new MoveGenericSubsystemWithPID(climberWinch, climberWinch.ENCODER_UP_LIMIT,
+                        climberWinch::getPosition, climberWinch.getPIDSettings(), climberWinch.getFFSettings()));
+
+        //stops the climber
         xbox.getBlueButton().whenPressed(new MoveGenericSubsystem(climberWinch, 0));
+
         xbox.getRedButton().whenPressed(new MoveGenericSubsystem(roller, 0));
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/OI.java:54]: add documentation

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

**Pull request closed without merging**